### ISSUE TITLE
connecting/wiring/plumbing SVG elements

### ISSUE THREAD

**Issue by arunkumar413** [adobe-webplatform/Snap.svg#493]
Hi Team,

JsPlumb has a function to establish connection between SVG elements using the below code. The connections are sustained even the SVG elements are moved/dragged. Can we have a similar functionality in snap.svg?

```
jsPlumb.connect({
  source:"element1", 
  target:"element2",
  anchors:["Right", "Left" ],
  endpoint:"Rectangle",
  endpointStyle:{ fill: "yellow" }
});
```

**Comment by ibrierley** [comment]
I did convert Dmitrys Raphael Graffle a while back, http://svg.dabbles.info/snaptut-graffle if its any use.

**Comment by arunkumar413** [comment]
HI Ian,

I had gone through the link and it seems that it can connect the Divs anywhere on the BBox of the element which is similar to the JsPlumb and would suffice for flow charts, graphs, tree and other logical  workflow related diagrams. 

Consider a scenario where we want to establish connections between electrical and electronic components. In this case we need to connect only the pins of the component for which we need to get the x.offset and y.offset of the pins. 

User should only define the connections between the pins (the pins can be css selectors or variables). 

As we would be connecting hundreds of components the code should be simple enough to connect the components and the logic to determine the offsets and drawing line/wire  should be taken care by the function. Below is the rough idea not the exact code of what I'm expecting 

var pin1 = s.select("#pin1");
var pin2 = s.select("#pin2")
connect(pin1, pin2);

function connect()  
{

var offsetx= pin.getoffsetx
var offsety=pin1.getoffsety
var offsetw =pin2.getoffsetx;
var offsetv=pin2.getoffsety
Draw line between the offsetx and offsetv

}

**Comment by ibrierley** [comment]
Is there a question in there ? Not quite sure this is a github issue, but
I'm not sure why you don't create a tiny svg object at the end of the pin
(it could even be invisible), and connect between those as an example ?

**Comment by arunkumar413** [comment]
Seems like a tiny svg object at the end of the pin would be a good idea and I'm trying this in my project.  Also the connecting lines in this code: http://svg.dabbles.info/snaptut-graffle become curved when the object is dragged. But I'd like to have a polyline connection that bends only at 90 degrees. May be I should tweak that code a bit.

**Comment by DavidBruchmann** [comment]
Maybe this is helping: https://github.com/neyric/wireit

**Comment by arunkumar413** [comment]
Hi David, 

I'm using a SVG line for connecting parts. It seems that the start and the end point of a line can be tied to the getbbox values of the part/connecting point. This way it appears that we can ensure that the wire/line remains intact even if the connection point/part is moved/dragged. 

Below is the implementation that's in my mind. Need to test it though. Modify the points between start and end points in such as way that the line is always orthogonal.

Also I realized that I raised this bug without having a full understanding of snapSVG power and capabilities. 

var cp1 = part1.getBBox();
var cp2 = part2.getBBox();

line = [cp1.x2, cp1.y, 20, 50, 60, 70, cp2.x2, cp2.y ];

Thanks,
Arun
